### PR TITLE
Config interface change

### PR DESCRIPTION
**Pull request by crufter**
Heavily wip

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -2,49 +2,107 @@
...
 )
 
+type Values interface {
+	Get(path string, options ...Option) Value
+	Set(path string, val interface{}, options ...Option)
+	Delete(path string, options ...Option)
+}
+
 // Config is an interface abstraction for dynamic configuration
 type Config interface {
```
Comment (ben-toogood) [inline config/config.go]: Can we just combine the two interfaces?

File: config/config.go [hunk 2]
```diff
@@ -2,49 +2,107 @@
...
-	Next() (reader.Value, error)
-	Stop() error
+	Values
 }
 
-type Options struct {
-	Loader loader.Loader
-	Reader reader.Reader
-	Source []source.Source
+// NewConfig returns new config
```
Comment (ben-toogood) [inline config/config.go]: I don't think we need this in the top level, can you please move to go-micro/config/store

**Review by ben-toogood — APPROVED** [review]

File: config/config.go (hunk 2 above)
Comment (crufter) [reply config/config.go]: I don't get this

File: config/config.go (hunk 1 above)
Comment (crufter) [reply config/config.go]: Yeah we can now

File: config/config.go (hunk 2 above)
Comment (ben-toogood) [reply config/config.go]: The implementation should't be in the top level package, only the interface. We'll import the implementations separately from within micro.

**Comment by crazybber** [comment]
any plan for file config implement ? @crufter @asim

**Pull request merged**